doc: Add the security overview (#64)
canonical/imagecraft: docs/explanation/security-overview.rst

.. _security-overview:

Security overview
==============================

> Overview of security aspects of imagecraft.

Reviewer: ```suggestion
Overview of security aspects of Imagecraft.
```

Privileged execution
--------------------
 
``imagecraft`` may require **elevated permissions** to properly run (when using the destructive mode). It is recommended to use a dedicated building machine. Make sure ``imagecraft`` is installed from a trusted source, and the provided configuration is trusted. 

Cryptography
------------

> ``imagecraft`` is a wrapper around several tools and mainly ``ubuntu-image``. The tool itself is not using cryptographic technologies. Please refer to the `ubuntu-image security overview`_ for details on cryptographic technologies usage in ``ubuntu-image``.

Reviewer: ```suggestion
``imagecraft`` is a wrapper around several tools and mainly ``ubuntu-image``. The tool itself does not use cryptographic technologies. Refer to the `ubuntu-image security overview`_ for details on cryptographic technologies used in ``ubuntu-image``.
```

Miscellaneous
~~~~~~~~~~~~~

* Secrets (passwords and hashes) can be present in the configuration files provided to ``imagecraft`` to build images. Specifically, in the ``ubuntu-bootstrap`` plugin configuration:

>   * In the ``extra-ppas`` customisation section, authentication tokens ``user:password`` can be defined to access private PPAs. These values are used to write the ``apt`` configuration without any treatment.

Reviewer: ```suggestion
  * In the ``extra-ppas`` customisation section, authentication tokens (``user:password``) can be defined to access private PPAs. These values are used to write the ``apt`` configuration without any treatment.
```
